### PR TITLE
sync-to-async

### PR DESCRIPTION
**Pull request by FelixZilber**

### CODE REVIEW

**Review by refael-m — COMMENTED** [review]
Comments

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -234,7 +238,7 @@ export namespace BrowserUtils {
    * @param options WaitForOptions options (optional) { timeout, timeoutMsg, interval }
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  export function waitUntil(condition: () => any, options?: WaitUntilOptions): any {
+  export async function waitUntil(condition: () => Promise<any>, options?: WaitUntilOptions): Promise<any> {
     return browser.waitUntil(() => condition(), { ...{ timeout: DEFAULT_TIME_OUT }, ...options });
```
Comment (refael-m) [inline src/commons/BrowserUtils.ts:242]: need to wait for condition, something like this
`    return browser.waitUntil(async () => await condition(), { ...{ timeout: DEFAULT_TIME_OUT }, ...options });
`

File: src/commons/BrowserUtils.ts [hunk 2]
```diff
@@ -259,25 +263,25 @@ export namespace BrowserUtils {
    * Return true or false if the selected DOM-element is enabled
    * @param selector - element selector
    */
-  export function isEnabled(selector: string): boolean {
-    Reporter.debug(`Is element enabled '${selector}'`);
+  export async function isEnabled(selector: string): Promise<boolean> {
+    await Reporter.debug(`Is element enabled '${selector}'`);
 
-    return $(selector).isEnabled();
+    return await $(selector).isEnabled();
```
Comment (refael-m) [inline src/commons/BrowserUtils.ts:269]: the 'await' her is unnecessary

File: src/commons/BrowserUtils.ts [hunk 3]
```diff
@@ -920,9 +930,9 @@ export namespace BrowserUtils {
    */
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  function tryBlock(action: () => any, errorMessage: string): any {
+  async function tryBlock(action: () => Promise<any>, errorMessage: string): Promise<any> {
     try {
-      return action();
+      return await action();
```
Comment (refael-m) [inline src/commons/BrowserUtils.ts:935]: the 'await' here is unnecessary

File: src/commons/AllureReport.ts [hunk 4]
```diff
@@ -39,30 +39,28 @@ export namespace AllureReporter {
    * Stop network audit by sending disable options to cdp method
    * ts-ignore used since it missing types support
    */
-  export function stopNetworkAudit(): void {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    browser.cdp('Network', 'disable');
+  export async function stopNetworkAudit(): Promise<void> {
+    await browser.cdp('Network', 'disable');
```
[the remark this answers is not in the document]
Comment (FelixZilber) [reply src/commons/AllureReport.ts:43]: there are additional action after this call

[the remark this answers is not in the document]
Comment (refael-m) [reply src/commons/AllureReport.ts:43]: I see the function ends right after, line 44, am I missing something?

[the remark this answers is not in the document]
Comment (FelixZilber) [reply src/commons/AllureReport.ts:43]: done

File: src/commons/BrowserUtils.ts (hunk 1 above)
Comment (FelixZilber) [reply src/commons/BrowserUtils.ts:242]: changed condition type to boolean | Promise<boolean>

File: src/commons/BrowserUtils.ts [hunk 5]
```diff
@@ -338,50 +354,50 @@ export namespace BrowserUtils {
...
    * The Get Window Handles command returns a list of window handles for every open top-level browsing context.
    * The order in which the window handles are returned is arbitrary.
    */
-  export function getWindowHandles(): Array<string> {
-    Reporter.debug('Get all ids of all open tabs');
+  export async function getWindowHandles(): Promise<Array<string>> {
+    await Reporter.debug('Get all ids of all open tabs');
 
-    return tryBlock(() => browser.getWindowHandles(), 'Failed to get window handles');
+    return await tryBlock(async () => await browser.getWindowHandles(), 'Failed to get window handles');
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:370]: according to `https://webdriver.io/docs/api/jsonwp/#getwindowhandles` this function is sync

File: src/commons/BrowserUtils.ts [hunk 6]
```diff
@@ -188,22 +201,22 @@ export namespace BrowserUtils {
   /**
    * Click browser's back button
    */
-  export function back(): void {
-    Reporter.debug('Click browser back button');
-    tryBlock(() => browser.back(), 'Failed to click browser back button');
+  export async function back(): Promise<void> {
+    await Reporter.debug('Click browser back button');
+    await tryBlock(async () => await browser.back(), 'Failed to click browser back button');
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:206]: browser.back is sync function

**Review by AlexeyCL — APPROVED** [review]

**Pull request merged**